apiserver/facades/controller/firewaller/firewaller.go

```diff
@@ -53,7 +51,7 @@ func NewStateFirewallerAPIV3(context facade.Context) (*FirewallerAPIV3, error) {
 	st := context.State()
 	environConfigGetter := stateenvirons.EnvironConfigGetter{st}
 	cloudSpecAPI := cloudspec.NewCloudSpec(environConfigGetter.CloudSpec, common.AuthFuncForTag(st.ModelTag()))
-	return NewFirewallerAPI(stateShim{st}, context.Resources(), context.Auth(), cloudSpecAPI)
+	return NewFirewallerAPI(stateShim{st: st, State: firewall.StateShim(st)}, context.Resources(), context.Auth(), cloudSpecAPI)
```

Comment:
Yeah, it's messy trying to share common code involving state shims from different packages. I though a bit if ick here was worth being able to try and keep the firewall state methods nicely together.